packages/react-instantsearch-hooks-dom/src/widgets/__tests__/InfiniteHits.test.tsx

```diff
@@ -0,0 +1,214 @@
+import { act, fireEvent, render, waitFor } from '@testing-library/react';
+import React from 'react';
+
+import { createSearchClient } from '../../../../../test/mock';
+import {
+  createMultiSearchResponse,
+  createSingleSearchResponse,
+} from '../../../../../test/mock/createAPIResponse';
+import { InstantSearchHooksTestWrapper, wait } from '../../../../../test/utils';
+import { InfiniteHits } from '../InfiniteHits';
+
+type CustomHit = {
+  somethingSpecial: string;
+};
+
+const searchClient = createSearchClient({
+  search: (requests) =>
+    Promise.resolve(
+      createMultiSearchResponse(
+        ...requests.map((request) => {
+          const { hitsPerPage = 3, page = 0 } = request.params!;
+          const hits = Array.from({ length: hitsPerPage }, (_, i) => {
+            const offset = hitsPerPage * page;
+            return {
+              objectID: (i + offset).toString(),
+              somethingSpecial: String.fromCharCode(
+                'a'.charCodeAt(0) + i + offset
+              ),
+            };
+          });
+
+          return createSingleSearchResponse<CustomHit>({
+            hits,
+            page,
+            nbPages: 10,
+            hitsPerPage,
+            index: request.indexName,
+          });
+        })
+      )
+    ),
+});
+
+describe('InfiniteHits', () => {
+  test('renders with default props', async () => {
+    const { container } = render(
+      <InstantSearchHooksTestWrapper searchClient={searchClient}>
+        <InfiniteHits />
+      </InstantSearchHooksTestWrapper>
+    );
+
+    await waitFor(() => {
+      expect(container.querySelector('.ais-InfiniteHits'))
+        .toMatchInlineSnapshot(`
+        <div
+          class="ais-InfiniteHits"
+        >
+          <ol
+            class="ais-InfiniteHits-list"
+          />
+          <button
+            class="ais-InfiniteHits-loadMore ais-InfiniteHits-loadMore--disabled"
+            disabled=""
+          >
+            Show more results
+          </button>
+        </div>
+      `);
+    });
+  });
+
+  test('renders with a non-default hit shape', async () => {
+    const { container } = render(
+      <InstantSearchHooksTestWrapper searchClient={searchClient}>
+        <InfiniteHits<CustomHit>
+          hitComponent={({ hit }) => (
+            <strong>{`${hit.__position} - ${hit.somethingSpecial}`}</strong>
+          )}
+        />
+      </InstantSearchHooksTestWrapper>
+    );
+
+    await waitFor(() => {
+      expect(container.querySelectorAll('strong')).toHaveLength(3);
+      expect(container.querySelector('.ais-InfiniteHits'))
+        .toMatchInlineSnapshot(`
+        <div
+          class="ais-InfiniteHits"
+        >
+          <button
+            class="ais-InfiniteHits-loadPrevious ais-InfiniteHits-loadPrevious--disabled"
+            disabled=""
+          >
+            Show previous results
+          </button>
```

Comment:
This should only be displayed if `showPrevious` is `true` or if we're loading the UI from page 2 or over, none of which are the case here, right?

Reply:
What I'm saying is that we shouldn't display this button by default. Right now it looks like it's visible all the time (although disabled when not relevant) which isn't consistent with InstantSearch.js.